app/assets/javascripts/oxalis/model/helpers/nml_helpers.js

```diff
@@ -251,6 +257,21 @@ function serializeComments(trees: Array<TreeType>): Array<string> {
   ];
 }
 
+function serializeTreeGroups(treeGroups: Array<TreeGroupType>): Array<string> {
+  return _.flatten(
+    treeGroups.map(treeGroup => {
+      const groupObject = { id: treeGroup.groupId, name: treeGroup.name };
+      return treeGroup.children.length > 0
+        ? [
+            serializeTag("group", groupObject, false),
+            ...indent(serializeTreeGroups(treeGroup.children)),
```

Comment:
I agree, I had to make a separate function though, because otherwise serializeTag would always need to return an array, which doesn't play well with how the function is used most of the time.